File: src/main/java/fluid/entity/DimensionalEntity.java
package fluid.entity;

import org.apache.commons.math3.linear.Array2DRowRealMatrix;

public interface DimensionalEntity {

    void setX(double x);

    double getX();

    void setY(double y);

    double getY();

    void setZ(double z);

    double getZ();

    double getDistance(DimensionalEntity other);

    static double getDistance(DimensionalEntity a, DimensionalEntity b) {
        return Math.sqrt(
                Math.pow((a.getX() - b.getX()), 2) +
                        Math.pow((a.getY() - b.getY()), 2) +
                        Math.pow((a.getZ() - b.getZ()), 2)
        );
    }

    Array2DRowRealMatrix getR4Matrix();
}

File: src/main/java/fluid/entity/DimensionalEntity.kt
package fluid.entity

import org.apache.commons.math3.linear.Array2DRowRealMatrix

interface DimensionalEntity {
    var x: Double
    var y: Double
    var z: Double
    fun getDistance(other: DimensionalEntity?): Double
    val r4Matrix: Array2DRowRealMatrix?

    companion object {
        @JvmStatic
        fun getDistance(a: DimensionalEntity, b: DimensionalEntity): Double {
            return Math.sqrt(
                    Math.pow(a.x - b.x, 2.0) +
                            Math.pow(a.y - b.y, 2.0) +
                            Math.pow(a.z - b.z, 2.0)
            )
        }
    }
}